File: 이것이 코딩테스트다/06.다이나믹 프로그래밍/4.개미전사.js
const texts = [4, [1, 3, 1, 5]];

const n = texts.shift();
let dp = Array(n).fill(0);
const graph = texts.pop();

dp[0] = graph[0];
dp[1] = Math.max(graph[0], graph[1]);

for (let i = 2; i < n; i++) {
  dp[i] = Math.max(dp[i - 2] + graph[i], dp[i - 1]);
}

console.log(dp[n - 1]);

File: 이것이 코딩테스트다/06.다이나믹 프로그래밍/4.개미전사.py
n = 4
food = [1,3,1,5]

dp = [0]*n
dp[0] = food[0]
dp[1] = max(food[1], food[0])
for i in range(2, n):
  dp[i] = max(dp[i-2]+food[i], dp[i-1])

print(max(dp))